ui/app/contexts/metametrics.new.js

```diff
@@ -128,72 +127,53 @@ export function MetaMetricsProvider({ children }) {
+    // have a matched route for tracking.
+    if (!match) {
+      // We have more specific pages for each type of transaction confirmation
+      // The user lands on /confirm-transaction first, then is redirected based on
+      // the contents of state.
+      if (location.pathname !== '/confirm-transaction') {
+        // Otherwise we are legitimately missing a matching route
         captureMessage(`Segment page tracking found unmatched route`, {
-          extra: {
-            previousMatch,
-            currentPath: location.pathname,
-          },
-        })
-      } else if (
-        previousMatch.current !== match.path &&
-        !(
-          environmentType === 'notification' &&
-          match.path === '/' &&
-          previousMatch.current === undefined
-        )
-      ) {
-        // When a notification window is open by a Dapp we do not want to track
-        // the initial home route load that can sometimes happen. To handle
-        // this we keep track of the previousMatch, and we skip the event track
-        // in the event that we are dealing with the initial load of the
-        // homepage
-        const { path, params } = match
-        const name = PATH_NAME_MAP[path]
-        segment.page({
-          [idTrait]: idValue,
-          name,
-          properties: {
-            // We do not want to send addresses or accounts in any events
-            // Some routes include these as params.
-            params: omit(params, ['account', 'address']),
-            locale: locale.replace('_', '-'),
-            network,
-            environment_type: environmentType,
-          },
-          context: {
-            ...context,
-            version,
-          },
+          previousMatch,
+          currentPath: location.pathname,
         })
       }
-      previousMatch.current = match?.path
+    } else if (
+      previousMatch.current !== match.path &&
+      !(
+        environmentType === 'notification' &&
+        match.path === '/' &&
+        previousMatch.current === undefined
+      )
+    ) {
+      // When a notification window is open by a Dapp we do not want to track the initial home route load that can
+      // sometimes happen. To handle this we keep track of the previousMatch, and we skip the event track in the event
+      // that we are dealing with the initial load of the homepage
+      const { path, params } = match
+      const name = PATH_NAME_MAP[path]
+      trackMetaMetricsPage(
+        {
+          name,
+          params: omit(params, ['account', 'address']),
+          environmentType,
+          page: context.page,
+          referrer: context.referrer,
+        },
+        {
+          isOptInPath: location.pathname.startsWith('/initialize'),
+        },
+      )
     }
-  }, [
-    location,
-    version,
-    locale,
-    context,
-    network,
-    metaMetricsId,
-    participateInMetaMetrics,
-  ])
+    previousMatch.current = match?.path
+  }, [location, dispatch, context])
```

Comment:
I can't see where `dispatch` is used :thinking:

Reply:
Left over from when this logic was in that thunk. removed!